doc/doxygen/headers/concepts.h

```diff
@@ -127,6 +105,68 @@
  * them manually.
  * </dd>
  *
+ * <dt class="concepts">@anchor ConceptMesh <b>Mesh</b></dt>
+ *
+ * <dd>
+ * Meshes can be thought of as arrays of vertices and connectivities, but a
+ * more fruitful view is to consider them as <i>collections of cells</i>. In
+ * C++, collections are often called <i>containers</i> (typical containers are
+ * std::vector, std::list, etc.) and they are characterized by the ability
+ * iterate over the elements of the collection. The <tt>Mesh</tt> concept
+ * refers to any container which defines appropriate methods (such as
+ * DoFHandler::begin_active()) and <tt>typedefs</tt> (such as
+ * DoFHandler::active_cell_iterator) for managing collections of cells.
+ *
+ * Triangulation&nnbsp;s, DoFHandler&nnbsp;s, and hp::DoFHandler&nnbsp;s may
```

Comment:
Funny, but not quite. `ZERO WIDTH SPACE` means that a word break can happen at that character location but otherwise the words will appear consecutively: see [the wikipedia article](https://en.wikipedia.org/wiki/Zero_width_space) for a nice demo with Lorem Ipsum.